specifications/bcastByz/bcastByz.tla

```diff
@@ -0,0 +1,983 @@
+    <2>6 IndInv_Unforg /\ Receive(i) => ~UponAcceptNotSent(i) 
+      <3> SUFFICES ASSUME IndInv_Unforg,
+                          Receive(i)
+                   PROVE ~UponAcceptNotSent(i)                   
+            OBVIOUS
+      <3>1 ~UponAcceptNotSent(i) =
+              \/ ~(pc[i] \in { "V0", "V1" } )
+              \/ ~(Cardinality(rcvd'[i]) >= N - T)              
+              \/ ~(pc' = [pc EXCEPT ![i] = "AC"])
+              \/ ~(sent' = sent \cup { <<i, "ECHO">> })
+              \/ ~(UNCHANGED << Corr, Faulty >>)
+        BY DEF UponAcceptNotSent  
+      <3>2 (Cardinality(rcvd'[i]) <= T)
+              => (Cardinality(rcvd'[i]) < N - T)
+        BY <2>3, NTFRel 
+      <3>3 ~(Cardinality(rcvd'[i]) >= N - T) 
+              = (Cardinality(rcvd'[i]) < N - T)
+        BY <2>3, NTFRel    
+      <3>4 (Cardinality(rcvd'[i]) <= T) 
+              => ~(Cardinality(rcvd'[i]) >= N - T)
+        BY <3>2, <3>3   
+      <3>5 (Cardinality(rcvd'[i]) <= T) => ~UponAcceptNotSent(i)
+        BY <3>1, <3>4              
+      <3>6 QED
+        BY <2>3, <3>5        
+    <2> IndInv_Unforg /\ Receive(i) =>  ~UponAccept(i)
+      <3> SUFFICES ASSUME IndInv_Unforg,
+                          Receive(i)
+                   PROVE ~UponAccept(i)                   
+            OBVIOUS
+      <3>1 ~UponAccept(i) =
+              \/ ~(pc[i] = "SE")
+              \/ ~(Cardinality(rcvd'[i]) >= N - T)              
+              \/ ~(pc' = [pc EXCEPT ![i] = "AC"])
+              \/ ~(sent' = sent)
+              \/ ~(UNCHANGED << Corr, Faulty >>)
+        BY DEF UponAccept  
+      <3>2 (Cardinality(rcvd'[i]) <= T)
+              => (Cardinality(rcvd'[i]) < N - T)
+        BY <2>3, NTFRel 
+      <3>3 ~(Cardinality(rcvd'[i]) >= N - T) 
+              = (Cardinality(rcvd'[i]) < N - T)
+        BY <2>3, NTFRel    
+      <3>4 (Cardinality(rcvd'[i]) <= T) 
+              => ~(Cardinality(rcvd'[i]) >= N - T)
+        BY <3>2, <3>3   
+      <3>5 (Cardinality(rcvd'[i]) <= T) => ~UponAccept(i)
+        BY <3>1, <3>4              
+      <3>6 QED
+        BY <2>3, <3>5        
+    <2>10 QED
+        BY <2>1, <2>2, <2>3, <2>4, <2>5, <2>6 DEF Step, IndInv_Unforg
+  <1>5 QED
+    BY <1>1, <1>2, <1>3, <1>4 DEF IndInv_Unforg    
+  
+THEOREM Unforg_Step3 == IndInv_Unforg => Unforg
+  <1>1 pc = [i \in Proc |-> "V0" ] => \A i \in Proc : pc[i] # "AC"
+    OBVIOUS
+  <1>2 (TypeOK /\ pc = [i \in Proc |-> "V0" ]) => \A i \in Proc : pc[i] # "AC"
+    BY <1>1
+  <1>3 (TypeOK /\ FCConstraints /\ pc = [i \in Proc |-> "V0" ]) => \A i \in Proc : pc[i] # "AC"
+    BY <1>2
+  <1>4 (TypeOK /\ FCConstraints /\ pc = [i \in Proc |-> "V0" ] /\ sent = {}) => \A i \in Proc : pc[i] # "AC"    
+    BY <1>3
+  <1>5 IndInv_Unforg => \A i \in Proc : pc[i] # "AC"   
+    BY <1>4 DEF IndInv_Unforg  
+  <1>6 IndInv_Unforg => \A i \in Proc : i \in Corr => pc[i] # "AC"
+    BY  <1>5     
+  <1>7 QED
+    BY <1>6 DEF Unforg 
+    
+THEOREM Unforg_Thm == SpecNoBcast => Unforg
```

Comment:
Again, make this a temporal theorem, using the previous theorems and temporal logic.